.github/workflows/ci.yml

```diff
@@ -0,0 +1,41 @@
+name: CI
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+jobs:
+  Tests:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        store: ["memory", "redis"]
+
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+
+    steps:
+      - name: Download source
+        uses: actions/checkout@v3
+
+      - name: Install Crystal
+        uses: crystal-lang/install-crystal@v1
+
+      - name: Cache shards
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/shards
+          key: shards-${{ hashFiles('shard.yml') }}
```

Comment:
shouldn't this be hashing the `shard.lock` file instead?

Reply:
Technically yes and no. I went with the "safe approach" [mentioned here](https://crystal-lang.org/reference/1.10/guides/ci/gh-actions.html#caching) but I can definitely change this to cache using `shard.lock`.

I guess that's actually better. This was based from another workflow I had built a while ago, where I didn't think much about it back then. I'm pushing the update in a minute 👍🏼